app/services/bulk_upload/lettings/row_parser.rb

```diff
@@ -137,49 +139,633 @@ class BulkUpload::Lettings::RowParser
+
+    attributes["housingneeds"] = housingneeds
+
+    attributes["illness"] = field_118
+
+    attributes["layear"] = field_66
+    attributes["waityear"] = field_67
+    attributes["reason"] = field_52
+    attributes["reasonother"] = field_53
+    attributes["prevten"] = field_61
+    attributes["homeless"] = homeless
+
+    attributes["prevloc"] = prevloc
+    attributes["previous_la_known"] = previous_la_known
+    attributes["ppcodenk"] = ppcodenk
+    attributes["ppostcode_full"] = ppostcode_full
+
+    attributes["reasonpref"] = field_69
+    attributes["rp_homeless"] = field_70
+    attributes["rp_insan_unsat"] = field_71
+    attributes["rp_medwel"] = field_72
+    attributes["rp_hardship"] = field_73
+    attributes["rp_dontknow"] = field_74
+
+    attributes["cbl"] = cbl
+    attributes["chr"] = chr
+    attributes["cap"] = cap
+    attributes["letting_allocation_unknown"] = letting_allocation_unknown
+
+    attributes["referral"] = field_78
+
+    attributes["net_income_known"] = net_income_known
+    attributes["earnings"] = earnings
+    attributes["incfreq"] = field_116
+    attributes["hb"] = field_48
+    attributes["benefits"] = field_49
+
+    attributes["period"] = field_79
+    attributes["brent"] = field_80
+    attributes["scharge"] = field_81
+    attributes["pscharge"] = field_82
+    attributes["supcharg"] = field_83
+    attributes["tcharge"] = field_84
+    attributes["chcharge"] = field_85
+    attributes["household_charge"] = field_86
+    attributes["hbrentshortfall"] = field_87
+    attributes["tshortfall_known"] = tshortfall_known
+    attributes["tshortfall"] = field_88
+
+    attributes["hhmemb"] = hhmemb
+
+    attributes["unitletas"] = field_105
+    attributes["rsnvac"] = field_106
+    attributes["sheltered"] = field_117
+
+    attributes["illness_type_1"] = field_119
+    attributes["illness_type_2"] = field_120
+    attributes["illness_type_3"] = field_121
+    attributes["illness_type_4"] = field_122
+    attributes["illness_type_5"] = field_123
+    attributes["illness_type_6"] = field_124
+    attributes["illness_type_7"] = field_125
+    attributes["illness_type_8"] = field_126
+    attributes["illness_type_9"] = field_127
+    attributes["illness_type_10"] = field_128
+
+    attributes["irproduct_other"] = field_131
+
+    attributes["offered"] = field_99
+
+    attributes["propcode"] = field_100
+
+    attributes["majorrepairs"] = majorrepairs
+
+    attributes["mrcdate"] = mrcdate
+
+    attributes["voiddate"] = voiddate
 
     attributes
   end
 
-  def field_4_presence_check
-    [1, 3, 5, 7, 9, 11].include?(field_1)
+  def voiddate
+    Date.new(field_91 + 2000, field_90, field_89) if field_91.present? && field_90.present? && field_89.present?
+  end
+
+  def majorrepairs
+    mrcdate.present? ? 1 : 0
+  end
+
+  def mrcdate
+    Date.new(field_94 + 2000, field_93, field_92) if field_94.present? && field_93.present? && field_92.present?
+  end
+
+  def prevloc
+    field_62
+  end
```

Comment:
does this need a method?

Reply:
well it's basically an `extract method` and gives an alias to what `field_62` actually is as it's been used elsewhere. I can remove if you feel strongly about it